### PR TITLE
add mandatory variables

### PR DESCRIPTION
**Pull request by callumrollo**
This PR adds a check for the mandatory variables. It also sets the three levels of severity to  `{3: "Mandatory", 2: "Highly Recommended", 1: "Suggested"}` as they are in OG1.0

### CODE REVIEW

File: cc_plugin_og/checker.py [hunk 1]
```diff
@@ -7,36 +7,47 @@
...
+            "PLATFORM_SERIAL_NUMBER",
+            "DEPLOYMENT_TIME",
+            "DEPLOYMENT_LATITUDE",
+            "DEPLOYMENT_LONGITUDE",
+        ]
+
+        out_of = len(required_variables)
+        for variable in required_variables:
+            try:
+                assert variable in dataset.variables
```
Comment (ocefpaf) [inline cc_plugin_og/checker.py:47]: We should not use Python `assert` outside of tests. It is dropped by JITs and other kinds of compilations. See https://docs.astral.sh/ruff/rules/assert/ for the lint explanation and some suggestions on how to replace it. Here we can do `if variable in dataset.variables: else ...`

PS: @benjwadams probably has a better way for these checks though, using existing compliance-checker machinery.

**Review by benjwadams — COMMENTED** [review]
There's a simple check harness for global attributes using this method: https://github.com/ioos/compliance-checker/blob/develop/compliance_checker/base.py#L415-L565

This is used in IOOS metadata checker, for example:  https://github.com/ioos/compliance-checker/blob/develop/compliance_checker/ioos.py#L544-L612

It's not usable with variable attributes presently, but could be adapted for this purpose if it made certain repetitive checks easier.

File: cc_plugin_og/checker.py (hunk 1 above)
Comment (benjwadams) [reply cc_plugin_og/checker.py:47]: `TestCtx` has methods that check for a particular named section and automatically adds the appropriate scoring values and messages depending on whether a test passes or fails: https://github.com/ioos/compliance-checker/blob/f543e4fa169635e31582b3804a7be06efa398db3/compliance_checker/base.py#L301-L358

I may rename it at some time as I'm using more Flask for some web applications and it has a "Test Context" that's used for unit tests.

**Comment by jcermauwedu** [comment]
My bad, I put the assert code was there as an example for @callumrollo -- I must have copied it from a `pytest` example.   It is also missing a class function to return a Result object.  I will add some things this evening with some live examples.

**Comment by jcermauwedu** [comment]
Commit 7283458 replaces the assert statements and implements the rule above.

**Pull request closed without merging**